pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -244,6 +244,17 @@ func validateStep(ctx context.Context, s Step, names sets.String) (errs *apis.Fi
 	return errs
 }
 
+// ValidateParameterSpecs validates that there are no duplicate names in the TaskSpec and all the types within a slice of ParamSpecs.
+func validateParameterSpecs(params []ParamSpec) (errs *apis.FieldError) {
```

Comment:
I think it might make more sense to add this validation to the function [ValidateParameterVariables](https://github.com/tektoncd/pipeline/blob/9486d4834be0b998cc54cedd631c7c1a45259b50/pkg/apis/pipeline/v1beta1/task_validation.go#L311), and likewise in [ValidatePipelineParameterVariables](https://github.com/tektoncd/pipeline/blob/c938ee5667fb64ddcf6e6c26e7732b68dde82104/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L128)

Reply:
Agreed! I made the necessary changes.

Reply:
It doesn't look like the changes were applied to ValidatePipelineParameterVariables?

Also, no need to make any changes to v1alpha1

Reply:
`validatePipelineParameterVariables` was already checking for [duplicated param names](https://github.com/tektoncd/pipeline/blob/a146455e9df70898a9d5b5891a78b5e27480ed7b/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L150-L152) and [param types](https://github.com/tektoncd/pipeline/blob/a146455e9df70898a9d5b5891a78b5e27480ed7b/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L140-L142) so no changed needed to be made there. It was only missing at the task validation stage.

Yes, now that I moved the function call to validateParamTypes back to where it was, `ctx` was no longer required by `ValidateParameterVariables` and so the change to `v1alpha1` was also undone.